Fix grammar and plural handling in action descriptions

NoRi2909 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

The three Soundtouch actions 
- create_zone
- add_zone_slave
- remove_zone_slave

all take a list of slave entity IDs as arguments. [See the online documentation](https://www.home-assistant.io/integrations/soundtouch/#action-create_zone).

This PR changes the descriptions in the UI to correctly reflect that along with replacing "master" / "slave" with "leader" / "follower".

Along with that this also fixes the grammar mistake in "Name of slaves entities" by replacing this with "Names of slave entities" and replaces "the … zone" with "a … zone" to make the corresponding action descriptions consistent.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

NoRi2909 committed:
Fix grammar and plural handling in action descriptions

The three Soundtouch actions 
- create_zone
- add_zone_slave
- remove_zone_slave

all take a list of slave entity IDs as arguments.

This commit changes the descriptions in the UI to correctly reflect that, matching the online documentation.

Along with that this also fixes the grammar mistake in "Name of slaves entities" by replacing this with "Names of slave entities" and replaces "the … zone" with "a … zone" to make the action descriptions consistent.

kroimon reviewed (approved):
Thank you!

joostlek reviewed (changes requested):
While we are at this? Could we migrate these lines to main/secondary, or leader/follower? Ideally we also update the code to do this, but I think the strings are a good first step here :)

kroimon commented:
I think the right way would be to implement the `media_player.join` and `media_player.unjoin` actions while deprecating the Soundtouch specific zone actions. 
As I don't have a Soundtouch system installed anymore, maybe you want to take this on, @NoRi2909?

NoRi2909 commented:
@joostlek Sure, why not.

@kroimon I'd use "leader / follower". What do you think?

NoRi2909 commented:
@kroimon I'm just into the German translations so I found this issue while going over the strings in Lokalise.
Code changes are well above what I can handle here.

So for the moment I can only offer to update this PR to the new terminology.

kroimon commented:
> @kroimon I'm just into the German translations so I found this issue while going over the strings in Lokalise. Code changes are well above what I can handle here.

Sure, no problem at all! Any contributions are very welcome :)

> @kroimon I'd use "leader / follower". What do you think?

Maybe you could adapt the descriptions of the [`media_player.join`](https://www.home-assistant.io/integrations/media_player/#action-media_playerjoin) and [`media_player.unjoin`](https://www.home-assistant.io/integrations/media_player/#action-media_playerunjoin) actions. They seem to completely avoid any explicit hierarchy terms. The action and attribute names will still use the master/slave terminology, but at least the descriptions might avoid that as far as possible.

NoRi2909 committed:
Use "media player" in descriptions, replace "master" / "slave"

This additional commit eliminates "master" and "slave" in the descriptions by just talking about "media player entities" and replaces them with "leader" and "follower" in the action and field names.

NoRi2909 commented:
I made the PR a compromise of both suggestions:

For all action and field names I replaced "master" / "slave" with "leader" / "follower".

In the descriptions I eliminated them by just referring to either as "media player entities".
This is also better than using "name" here as it's the entity_ID in YAML and a selector in the UI (I assume).

As this will be deprecated then soon I think that should be enough as a bridge to a complete overhaul.

NoRi2909 committed:
Replace 2 occurrences of "master" key with references

NoRi2909 commented:
In addition I replaced two "master" keys with references to the first action, making this consistent with the handling of the "slaves" keys.

davidrapan reviewed (approved):
I think it's funny to rename those controversial terms in the code while still having master branch (even so I have very different opinion about this all together and on top of that I like the parent-child form more but maybe it has some deeper meaning here?) but 👍.

joostlek commented:
@davidrapan It's also not that we are now rushing to get every one of those words out, but we do try to prevent new ones from getting in the code base and just like here, if we happen to touch code where we still use it, we might as well fix it along the way.
